rules/TypeDeclarationDocblocks/Rector/Class_/DocblockVarArrayFromPropertyDefaultsRector.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\TypeDeclarationDocblocks\Rector\Class_;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\Property;
+use PHPStan\PhpDocParser\Ast\PhpDoc\VarTagValueNode;
+use PHPStan\Type\Type;
+use Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfo;
+use Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfoFactory;
+use Rector\Comments\NodeDocBlock\DocBlockUpdater;
+use Rector\Privatization\TypeManipulator\TypeNormalizer;
+use Rector\Rector\AbstractRector;
+use Rector\StaticTypeMapper\StaticTypeMapper;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\TypeDeclarationDocblocks\Rector\Class_\DocblockVarArrayFromPropertyDefaultsRector\DocblockVarArrayFromPropertyDefaultsRectorTest
+ */
+final class DocblockVarArrayFromPropertyDefaultsRector extends AbstractRector
+{
+    public function __construct(
+        private readonly PhpDocInfoFactory $phpDocInfoFactory,
+        private readonly StaticTypeMapper $staticTypeMapper,
+        private readonly DocBlockUpdater $docBlockUpdater,
+        private readonly TypeNormalizer $typeNormalizer,
+    ) {
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Add @var array docblock to array property based on iterable default value', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    private array $items = [1, 2, 3];
+}
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    /**
+     * @var int[]
+     */
+    private array $items = [1, 2, 3];
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        foreach ($node->getProperties() as $property) {
+            if (! $property->type instanceof Identifier) {
+                continue;
+            }
+
+            if (! $this->isName($property->type, 'array')) {
+                continue;
+            }
+
+            if (count($property->props) > 1) {
+                continue;
+            }
+
+            $soleProperty = $property->props[0];
+            if (! $soleProperty->default instanceof Array_) {
+                continue;
+            }
+
+            $propertyDefaultType = $this->getType($soleProperty->default);
+
+            $propertyPhpDocInfo = $this->phpDocInfoFactory->createFromNodeOrEmpty($property);
+
+            // type is already known
+            if ($propertyPhpDocInfo->getVarTagValueNode() instanceof VarTagValueNode) {
+                continue;
+            }
+
+            $this->addGeneratedTypeReturnDocblockType($propertyDefaultType, $propertyPhpDocInfo, $property);
```

Comment:
the doc is based on default, the method name need to be renamed, eg:

```suggestion
            $this->addGeneratedTypePropertyBasedOnDefault$propertyDefaultType, $propertyPhpDocInfo, $property);
```

Reply:
Outsourced :+1: